src/systems/physics-system.js

```diff
@@ -0,0 +1,101 @@
+import { World } from "three-ammo";
+import { TYPE } from "three-ammo/constants";
+
+const WORLD_CONFIG = {
+  debugDrawMode: THREE.AmmoDebugConstants.DrawWireframe,
+  gravity: { x: 0, y: -9.8, z: 0 }
+};
+
+export class PhysicsSystem {
+  constructor(scene) {
+    const Ammo = require("ammo.js/builds/ammo.wasm.js");
+    const AmmoWasm = require("ammo.js/builds/ammo.wasm.wasm");
+    const AmmoModule = Ammo.bind(undefined, {
+      locateFile(path) {
+        if (path.endsWith(".wasm")) {
+          return AmmoWasm;
+        }
+        return path;
+      }
+    });
+
+    this.bodies = [];
+    this.bodyHelpers = [];
+    this.shapeHelpers = [];
+
+    this.debugRequested = false;
+    this.debugEnabled = false;
+    this.scene = scene;
+    this.stepDuration = 0;
+
+    AmmoModule().then(() => {
+      this.world = new World(WORLD_CONFIG);
+    });
+  }
+
+  setDebug(debug) {
+    this.debugRequested = debug;
+  }
+
+  tick(dt) {
+    if (this.world) {
+      if (this.debugRequested !== this.debugEnabled) {
+        this.debugEnabled = this.debugRequested;
+        if (this.debugEnabled) {
+          this.world.getDebugDrawer(this.scene).enable();
+        } else {
+          this.world.getDebugDrawer(this.scene).disable();
+        }
+      }
+
+      while (this.bodyHelpers.length > 0) {
+        this.bodyHelpers.shift().init2();
+      }
+      while (this.shapeHelpers.length > 0) {
+        this.shapeHelpers.shift().init2();
+      }
```

Comment:
Is there a reason to do it this way instead of
```javascript
for (const bodyHelper of bodyHelpers) {
  bodyHelper.init2();
}
for (const shapeHelper of shapeHelpers) {
  shapeHelper.init2();
}
this.shapeHelpers = [];
this.bodyHelpers =[];
```
My way looks a little more normal to me.

Reply:
Is there a reason to do this in `tick`, instead of doing it when Ammo is loaded and the `world` is created?

Reply:
In your example I would personally do:
```
this.shapeHelpers.length = 0;
this.bodyHelpers.length = 0;
```
But I find little difference doing it with a while loop vs a for loop. I agree though this can be moved into the promise callback where `world` is created.